### PR TITLE
add subscription diagnostic logs

### PR DESCRIPTION
**Pull request by cristipogacean**
* log revised values for settings of subscriptions and monitored items
* log warnings when sequence numbers are missed

### CODE REVIEW

**Review by monikavar — DISMISSED** [review]

**Review by marcschier — DISMISSED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs [hunk 1]
```diff
@@ -770,19 +817,26 @@ internal sealed class SubscriptionWrapper : ISubscription {
...
                         GetValueOrDefault(DateTime.UtcNow);
 
-                    _logger.Debug("DataChange for subscription '{subscription}'/'{sessionId}', sequence#: "
-                        + "{Sequence} isKeepAlive: {KeepAlive}, publishTime: {PublishTime}",
-                        Id,
-                        Connection.CreateConnectionId(),
-                        sequenceNumber,
-                        isKeepAlive,
-                        publishTime);
+                    if (_currentSequenceNumber == sequenceNumber) {
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs]: Seems like `_expectedSequenceNumber` would be a better name than `_currentSequenceNumber` as it is not the one that we have received before but the one that we expect now. What  do you think ?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs [hunk 2]
```diff
@@ -770,19 +817,26 @@ internal sealed class SubscriptionWrapper : ISubscription {
...
+                        _logger.Debug("DataChange for subscription '{subscription}'/'{sessionId}', sequence#: "
+                            + "{Sequence} isKeepAlive: {KeepAlive}, publishTime: {PublishTime}",
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs]: Maybe we can have a `,` after `{Sequence}` ?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs [hunk 3]
```diff
@@ -660,84 +681,110 @@ internal sealed class SubscriptionWrapper : ISubscription {
...
                         // do not deactivate an already activated subscription
                         subscription.SetPublishingMode(activate);
+                        logRevisedValues = true;
                     }
                 }
+
+                if (logRevisedValues && activate) {
+                    var diagInfo = new StringBuilder();
+                    diagInfo.Append("Subscription '{subscription}'/'{sessionId}' state actual(revised)/desired: ");
+                    diagInfo.Append("PublishingEnabled {currentPublishingEnabled}/{publishingEnabled} ");
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs]: Should we add commas after reporting expected and actual values for each field ?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs]: agreed!

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs (hunk 2 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs]: indeed

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs (hunk 3 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs]: sure

**Review by karok2m — APPROVED** [review]

**Pull request merged**